bin/setup.csh

```diff
@@ -0,0 +1,46 @@
+# setup lsstsw environment
+#
+# source this file from your ~/.cshrc
+#
+# possibly blank, but can't be parsed as $_
+set cmd=($_)
+if ( $?0 ) then
+    set source=$0
+else if ( $#cmd >= 3 ) then
+    set source=`pwd`/${cmd[2]}
+else if (-f /usr/sbin/lsof ) then
+    set source=`/usr/sbin/lsof +p $$ | grep -oE /.\*setup.csh`
```

Comment:
`newinstall` is run from the LSST root directory (once?). Both `setup.sh` and the new scripts are designed to set up environment variables automatically when the user logs in, so it can't rely on the user executing from a specific location on the build path.

So that entire ugly tree is the best csh emulation I can find for `$BASH_SOURCE`